### DISCUSSION TITLE
Upstream DNS servers - using unbound - ipv6 syntax on nonstandard port?

### DISCUSSION THREAD

**Discussion by cjustinhall** [AdguardTeam/AdGuardHome#2827] [Q&A]
Running adguard home and unbound together on a rhel8 box. Port 53 for dns/adugard and 5335 for unbound. Adguard home + unbound is working great for ipv4. The RHEL box and unbound (using dig) all work for ipv6 at the command line. Also, adguard home with public ipv6 dns upstream servers work.

**Here is my question. What is the correct syntax for a the ipv6 unbound server locally on port 5335?**  Tried ::1:5335 and other things. Just keep getting the error: Server "::1#5335": could not be used, please check that you've written it correctly.

#ipv4 unbound server locally - works fine
127.0.0.1:5335
#ipv6 public server - works fine
2620:fe::fe
**#here is my question! what should this format be???**
0::1#5335

**Comment by ameshkov** [comment]
Since your unbound listens to `::` (i.e. it listens to ALL network interfaces) you can use any IP address that your server has. Probably, `127.0.0.1` would be okay.

**Comment by Rar9** [comment]
I'm comming from pihole + unbound 

So difference is the port declaration # (pihole) vs : (adguard)

Ipv4 127.0.0.1:5335 works
But
Ipv6 ::1:5335   is also not working for me. 

It would be much appreciated if some working guide can be provided for normal installation or docker installation. 

In addition in combination with unbound should the adguard dns cache be disabled as we already use unbounds cache.

Whats recommended?

**Reply by ameshkov** [reply]
> Ipv6 ::1:5335 is also not working for me.

`[::1]:5335` - this is the proper format for IPv6 addresses

> In addition in combination with unbound should the adguard dns cache be disabled as we already use unbounds cache.

In any case, I'd better keep AdGuard Home cache enabled, this way you'll avoid making extra connections to unbound when they're not needed.

**Reply by Rar9** [reply]
Thanks for the fast answer.

it now saves, with
127.0.0.1:5335
[::1]:5335

but testing upstream fails (normal ubuntu 20.04 install - no docker) 

**Server "[::1]:5335": could not be used, please check that you've written it correctly**

**Reply by ameshkov** [reply]
Well, it's possible that it does not listen there. Why do you need it anyway? There's no point in using different addresses of the very same unbound instance.
